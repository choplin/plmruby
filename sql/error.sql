CREATE FUNCTION plmruby_raise() RETURNS int AS
$$
	raise 'foo'
$$
LANGUAGE plmruby IMMUTABLE STRICT;
SELECT plmruby_raise();

CREATE FUNCTION plmruby_no_method_error() RETURNS int AS
$$
	'foo'.bar
$$
LANGUAGE plmruby IMMUTABLE STRICT;
SELECT plmruby_no_method_error();

DO $$ raise 'foo' $$ LANGUAGE plmruby;
DO $$ 'foo'.bar $$ LANGUAGE plmruby;
