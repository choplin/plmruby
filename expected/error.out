CREATE FUNCTION plmruby_raise() RETURNS int AS
$$
	raise 'foo'
$$
LANGUAGE plmruby IMMUTABLE STRICT;
SELECT plmruby_raise();
ERROR:  RuntimeError: foo
CREATE FUNCTION plmruby_no_method_error() RETURNS int AS
$$
	'foo'.bar
$$
LANGUAGE plmruby IMMUTABLE STRICT;
SELECT plmruby_no_method_error();
ERROR:  NoMethodError: undefined method 'bar' for "foo"
DO $$ raise 'foo' $$ LANGUAGE plmruby;
ERROR:  RuntimeError: foo
DO $$ 'foo'.bar $$ LANGUAGE plmruby;
ERROR:  NoMethodError: undefined method 'bar' for "foo"
